src/openms/source/ANALYSIS/OPENSWATH/DIAHelper.cpp

```diff
@@ -328,36 +327,68 @@ namespace OpenMS::DIAHelpers
       }
     }
 
-    // for SWATH -- get the theoretical b and y series masses for a sequence
-    void getBYSeries(const AASequence& a, //
-                     std::vector<double>& bseries, //
-                     std::vector<double>& yseries, //
+    // for SWATH -- get the theoretical prefix and suffix series masses for a sequence
+    void getPrefixSuffixSeries(const AASequence& a, //
+                     std::vector<double>& prefix_series, //
+                     std::vector<double>& suffix_series, //
                      TheoreticalSpectrumGenerator const * generator,
                      int charge)
     {
       // Note: We pass TheoreticalSpectrumGenerator ptr, as constructing it each time is too slow.
-      OPENMS_PRECONDITION(charge > 0, "For constructing b/y series we require charge being a positive integer");
+      OPENMS_PRECONDITION(charge > 0, "For constructing prefix/suffix series we require charge being a positive integer");
 
       if (a.empty()) return;
 
       PeakSpectrum spec;
       generator->getSpectrum(spec, a, charge, charge);
 
-      // Data array is present if AASequence is not empty
+      // Data array is present if Sequence is not empty
       const PeakSpectrum::StringDataArray& ion_name = spec.getStringDataArrays()[0];
-
+      // TODO is it faster to just pass two generators, one for prefix and one for suffix?
```

Comment:
With the generators that Eugen developed for XL mass spec I would say yes (they have a FAST version). I am also not sure if it is strictly ideal to provide the TheoreticalSpectrumGenerator here because it could be wrongly configured (e.g. to generate only a  and z ions, etc.).

Reply:
not sure. I think b and y is hardcoded all around OpenSWATH. Not sure if ETD DIA is a thing in proteomics

Reply:
When I did some speed benchmarks a while ago TheoreticalSpectrumGenerator is one of the speed bottlenecks for OpenSWATH so optimizing it would be good. 

I don't think OpenSWATH or DIA is being run on ETD data.